src/bf_dpu_update.py

```diff
@@ -889,6 +889,78 @@ def get_bmc_rshim_misc(self):
         ))
         return misc
 
+    def query_golden_image_config_dir_exists_on_bmc(self):
+        out = None
+        try:
+            out = self.run_command_on_bmc("sshpass -p {password} {ssh} {username}@{ip} '{command}'".format(
+                ssh=self.ssh,
+                password=self.ssh_password,
+                username=self.ssh_username,
+                ip=self.bmc_ip,
+                command='/bin/bash -c "[ -d /tmp/golden-image-config ] && echo YES || echo NO"',
+                exit_on_error=False
+            ))
+        except Exception as e:
+            print("Query BMC Config DIR failed: {0}".format(str(e)))
+            return False
+
+        if isinstance(out, dict) and "stdout" in out:
+            text = str(out["stdout"]).strip()
+        else:
+            text = str(out).strip()
+        
+        print("Query BMC Config DIR: {}".format(text))
+        return text.endswith("YES")
+    
+    def restart_golden_image_service_on_bmc(self):
+        try:
+            print("TRY restart golden-image service on BMC")
+            out = self.run_command_on_bmc("sshpass -p {password} {ssh} {username}@{ip} '{command}'".format(
+                ssh=self.ssh,
+                password=self.ssh_password,
+                username=self.ssh_username,
+                ip=self.bmc_ip,
+                command='/bin/bash -c "systemctl restart nvidia-golden-image-updater.service"',
+                exit_on_error=False
+            ))
+        except Exception as e:
+            print("restart nvidia-golden-image-updater.service failed: {0}".format(str(e)))
+            return False
+        
+        return True
+
+    def ensure_golden_image_config_dir_on_bmc(self, max_retries=6, sleep_secs=5):
+        """
+        Ensure /tmp/golden-image-config exists on BMC with up to `max_retries` attempts.
+        Each attempt:
+        1) query dir existence
+        2) 6 try: reboot BMC + wait_update_service_ready
+        3) query dir existence again
+        Return True on success, else raise Err_Exception on final failure.
+        """
+        for attempt in range(1, max_retries + 1):
+            # query DIR
+            if self.query_golden_image_config_dir_exists_on_bmc():
+                return True
+
+            try:
+                print("Trying to recover the golden-image-config DIR by BMC reboot: round {}".format(attempt))
+                self.reboot_bmc()
+                self.wait_update_service_ready()
+                time.sleep(sleep_secs)
+            except Exception:
+                pass
+
+            # query agian if DIR come back
```

Comment:
These 3 lines can be removed, you have the check in the beginning of the loop. Or you can move the checking outside after the for loop if needed.